packages/ui-components/src/components/flow-canvas/canvas-context.tsx

```diff
@@ -17,6 +32,8 @@ export const CanvasContextProvider = ({
   children: ReactNode;
 }) => {
   const [panningMode, setPanningMode] = useState<PanningMode>('grab');
+  const [selectedNodes, setSelectedNodes] = useState<WorkflowNode[]>([]);
```

Comment:
do we need to store the full WorkflowNode or only the step name?
Also, is there a problem that we don't update the selectedNodes when the workflow changes? (example on deletes)